docs/tutorial/sql.md

```diff
@@ -1195,6 +1195,66 @@ Output:
 +------------------------------+--------+--------------------------------------------------+-----------------+
 ```
 
+## Spatial vectorized udfs (Python only)
+
+By default when you create the user defined functions in Python, the UDFs are not vectorized.
+This means that the UDFs are called row by row which can be slow.
+To speed up the UDFs, you can use the `vectorized` UDF which will be called in a batch mode
+using Apache Arrow.
+
+To create a vectorized UDF please use the decorator sedona_vectorized_udf.
+Currently supports only the scalar UDFs. Vectorized UDFs are way faster than
+the normal UDFs. It might be even 2x faster than the normal UDFs.
+
+Decorator signature looks as follows:
+
+```python
+def sedona_vectorized_udf(udf_type: SedonaUDFType = SedonaUDFType.SHAPELY_SCALAR)
+```
+
+where udf_type is the type of the UDF function, currently supported are:
+
+- SHAPELY_SCALAR
+- GEO_SERIES
+
+The main difference is what input data you get in the function
+Let's analyze the two examples below, that creates buffers from
+a given geometry.
+
+### Shapely scalar UDF
+
+```python
+import shapely.geometry.base as b
+from sedona.sql.functions import sedona_vectorized_udf, SedonaUDFType
+
+@sedona_vectorized_udf()
+def vectorized_buffer(geom: b.BaseGeometry) -> b.BaseGeometry:
+    return geom.buffer(0.1)
+```
+
+### GeoSeries UDF
+
+```python
+import geopandas as gpd
+from sedona.sql.functions import sedona_vectorized_udf, SedonaUDFType
+
+@sedona_vectorized_udf(udf_type=SedonaUDFType.GEO_SERIES)
+def vectorized_geo_series_buffer(series: gpd.GeoSeries) -> gpd.GeoSeries:
+    buffered = series.buffer(0.1)
+
+    return buffered
+```
```

Comment:
Do we support other variants of UDFs involving geometries?

* UDFs taking geometry as input and returning a numeric value
* UDFs taking a numeric value and returning a geometry object
* UDFs that has more than 1 parameters, has geometry as parameter type or return type

Reply:
Let me verify those; my plan is to add new functionalities later, like table functions or agg functions.

Reply:
Looks I messed up a little, now I fixed it and tested

Reply:
I am not sure if this is going to work, as more than 1 column is not scala, and I didn't intend to add this in this MR; I am planning to add this later.